edr_server/core/models/parameters.py

```diff
@@ -49,10 +49,13 @@ def _prepare_json_for_init(cls, json_dict: JsonDict) -> JsonDict:
         if "symbol" in json_dict and isinstance(json_dict["symbol"], dict):
             json_dict["symbol"] = Symbol.from_json(json_dict["symbol"])
 
-        if "labels" in json_dict and isinstance(json_dict["label"], dict):
+        if "label" in json_dict and isinstance(json_dict["label"], dict):
             # Note, the field is called `label` in the serialised output, even though it can hold multiple values
             json_dict["label"] = LanguageMap.from_json(json_dict["label"])
 
+        if "label" in json_dict and isinstance(json_dict["label"], str):
+            json_dict["labels"] = json_dict.pop('label')
```

Comment:
typo? Plural `labels` in the `json_dict["labels"]`?

Reply:
Thanks, the idea is that lines 56 and 57 can still run after 52-54 but should probably test that!

Reply:
I'm not sure that's going to do what you expect, since if line 54 runs then the `labels` field will contain a `LanguageMap` object rather than `str` :)

It'd be a good idea to add a unit test for the handling of the `LanguageMap` case, since this labels field is confusing

Reply:
Have added fix and accompanying test.